glean-core/ios/Glean/Config/Configuration.swift

```diff
@@ -3,16 +3,54 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 struct Configuration {
-    let serverEndpoint: String = Configuration.DEFAULT_TELEMETRY_ENDPOINT
-    let userAgent: String = Configuration.DEFAULT_USER_AGENT
-    let connectionTimeout: Int = Configuration.DEFAULT_CONNECTION_TIMEOUT
-    let readTimeout: Int = Configuration.DEFAULT_READ_TIMEOUT
-    let logPings: Bool = Configuration.DEFAULT_LOG_PINGS
-    // let httpClient: HttpPingUploader = HttpPingUploader()
-    let maxEvents: Int? = nil
-    let pingTag: String? = nil
-    let channel: String? = nil
+    let serverEndpoint: String
+    let userAgent: String
+    let connectionTimeout: Int
+    let readTimeout: Int
+    let logPings: Bool
+    let httpClient: HttpPingUploader
+    let maxEvents: Int?
+    let pingTag: String?
+    let channel: String?
 
+    internal init(
```

Comment:
Can you please add a bit more docs about this? It's ok to copy stuff from the Android implementation, if needed.

Reply:
Will do